### PR TITLE
Add twitter bot error check when getting tar content

### PR DESCRIPTION
**Pull request by jpmonettas**

### CODE REVIEW

File: src/memefactory/server/twitter_bot.cljs [hunk 1]
```diff
@@ -71,18 +71,22 @@
...
+             tmp-name (str (random-uuid))
+             tar-file (str tmp-dir "/" tmp-name ".tar")
+             img-tmp-dir (str tmp-dir "/" tmp-name)
+             extract-stream (.extract tar-fs img-tmp-dir)]
+         (.writeFileSync fs tar-file buffer)
+         (.pipe (.createReadStream fs tar-file) extract-stream)
+         (.on extract-stream "finish"
+              (fn []
+                (resolve (.readFileSync fs (str img-tmp-dir "/" (aget (.readdirSync fs img-tmp-dir) 0)))))))
+       (catch js/Error e
```
Comment (fbielejec) [inline src/memefactory/server/twitter_bot.cljs]: catch :default e

File: src/memefactory/server/twitter_bot.cljs [hunk 2]
```diff
@@ -95,12 +99,12 @@
          (-> (server-utils/get-ipfs-binary-file ipfs-hash)
              (.then (fn [image-tar-file-content]
                       (when-not just-log-tweet?
-                        (.then (first-tar-obj image-tar-file-content)
-                               (fn [image-file-content]
-                                 (-> (upload-file-to-twitter twitter-obj image-file-content)
-                                     (.then (fn [media-id] (resolve media-id)))
-                                     (.catch (fn [e] (log/error "Error uploading media to twitter" {:registry-entry registry-entry
-                                                                                                    :ipfs-hash ipfs-hash})))))))))))))))
+                        (-> (first-tar-obj image-tar-file-content)
```
Comment (fbielejec) [inline src/memefactory/server/twitter_bot.cljs]: use promise-> macro

File: src/memefactory/server/twitter_bot.cljs (hunk 1 above)
Comment (jpmonettas) [reply src/memefactory/server/twitter_bot.cljs]: What is the difference?

File: src/memefactory/server/twitter_bot.cljs (hunk 2 above)
Comment (jpmonettas) [reply src/memefactory/server/twitter_bot.cljs]: also what is the difference here now we have the default rejection handler?

Comment (fbielejec) [reply src/memefactory/server/twitter_bot.cljs]: Preference - avoid boilerplate code, error checking and logging handled. YMMV

File: src/memefactory/server/twitter_bot.cljs (hunk 1 above)
Comment (fbielejec) [reply src/memefactory/server/twitter_bot.cljs]: JS allows to throw everything (also js/Object), this will catch everything

**Pull request merged**